client/app/dbaas/logs/detail/options/logs-options.service.js

```diff
@@ -0,0 +1,136 @@
+        const optionConfig = {};
+        optionConfig.type = this.$translate.instant(`${option}-type`);
+        optionConfig.detail = this.$translate.instant(`${option}-detail`);
+        optionConfig.quantity = optionsCountMap[option];
+        return optionConfig;
+    }
+
+    /**
+     * returns the list of options that have been subscribed in the service
+     *
+     * @param {any} serviceName
+     * @returns promise that resolves with the array of options which have been subscribed
+     * @memberof LogsOptionsService
+     */
+    getSubscribedOptions (serviceName) {
+        const transformSubscribedOption = this.transformSubscribedOption.bind(this);
+        return this.OvhApiDbaasLogs.Accounting().Aapi().me({
+            serviceName
+        }).$promise
+            .then(response => {
+                const optionsCountMap = _.reduce(response.options, (optionsMap, option) => {
+                    optionsMap[option.reference] = optionsMap[option.reference] ? ++optionsMap[option.reference] : 1;
+                    return optionsMap;
+                }, {});
+
+                return _.map(_.keys(optionsCountMap), option => transformSubscribedOption(option, optionsCountMap));
+            })
+            .catch(this.ServiceHelper.errorHandler("logs_options_current_options_loading_error"));
+    }
+
+    /**
+     * returns the list of options that have to be ordered (quantity > 0)
+     *
+     * @param {any} serviceName
+     * @returns the list of options to be ordered
+     * @memberof LogsOptionsService
+     */
+    getOptionsToOrder (options) {
+        return _.filter(options, option => option.quantity > 0);
+    }
+
+    /**
+     * returns the transformed option. Meant to be used to construct the order URL
+     *
+     * @param {any} option
+     * @returns the transformed option
+     * @memberof LogsOptionsService
+     */
+    transformOptionForOrder (option, serviceName) {
+        return {
+            planCode: option.planCode,
+            quantity: option.quantity,
+            serviceName,
+            productId: this.LogOptionConstant.productName
+        };
+    }
+    /**
+     * returns the order url that can be used to order the options
+     *
+     * @param {any} serviceName
+     * @returns the order url. This url has the selected options as arguments
+     * @memberof LogsOptionsService
+     */
+    getOrderURL (options, serviceName) {
+        const transformOptionForOrder = this.transformOptionForOrder.bind(this);
+        const products = this.getOptionsToOrder(options)
+            .map(option => transformOptionForOrder(option, serviceName));
+
+        let orderBaseUrl;
+        if (this.$window.location.host === this.LogOptionConstant.USHost) {
+            orderBaseUrl = this.LogOptionConstant.orderBaseURL.US;
+        }
+        orderBaseUrl = orderBaseUrl || this.LogOptionConstant.orderBaseURL.restOfWorld;
+
+        return "{orderBaseUrl}?products={products}"
```

Comment:
use OrderHelper instead :).  It constructs an express order URL based on a client subsidiary + an hash of parameters

Reply:
This is done :)